examples/adder/tests/test_adder.py

```diff
@@ -1,47 +1,38 @@
 # Simple tests for an adder module
 import cocotb
 from cocotb.triggers import Timer
-from cocotb.result import TestFailure
 from adder_model import adder_model
 import random
 
 
 @cocotb.test()
-def adder_basic_test(dut):
+async def adder_basic_test(dut):
     """Test for 5 + 10"""
-    yield Timer(2, units='ns')
+
     A = 5
     B = 10
 
-    dut.A = A
-    dut.B = B
+    dut.A <= A
+    dut.B <= B
 
-    yield Timer(2, units='ns')
+    await Timer(2, units='ns')
 
-    if int(dut.X) != adder_model(A, B):
-        raise TestFailure(
-            "Adder result is incorrect: %s != 15" % str(dut.X))
-    else:  # these last two lines are not strictly necessary
-        dut._log.info("Ok!")
+    assert int(dut.X) == adder_model(A, B), "Adder result is incorrect: {} != 15".format(dut.X)
```

Comment:
Prefer `assert` over raising a `TestFailure` and use test end to mark success rather than raising `TestSuccess`.